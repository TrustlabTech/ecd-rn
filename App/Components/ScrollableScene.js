import React, { Component } from 'react'
import {
  ScrollView,
  View,
  ActivityIndicator
} from 'react-native'
import Scene from '../Components/Scene'
import { Colours } from '../GlobalStyles'

export default class ScrollableScene extends Component {


  constructor(props) {
    super(props)
  }

  render() {
    if(this.props.loaded) {

      return (

        <ScrollView style={{flex: 1, backgroundColor: Colours.sceneBackgroundColour }}>
          {this.props.children}
        </ScrollView>
      )
    } else {
      return (
        <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
          <ActivityIndicator
            animating={true}
            style={{height: 80}}
            size="large"
          />
        </View>
      )
    }
  }
}
