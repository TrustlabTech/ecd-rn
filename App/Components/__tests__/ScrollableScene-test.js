import 'react-native'
import React from 'react'
import ScrollableScene from '../ScrollableScene'

import renderer from 'react-test-renderer'

it('renders correctly loaded', () => {
  const tree = renderer.create(
    <ScrollableScene loaded={true}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly not loaded', () => {
  const tree = renderer.create(
    <ScrollableScene loaded={false}/>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})