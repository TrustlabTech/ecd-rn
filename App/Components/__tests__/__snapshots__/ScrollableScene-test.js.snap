exports[`test renders correctly loaded 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "#fefefe",
      "flex": 1,
    }
  } />
`;

exports[`test renders correctly not loaded 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "height": 80,
      }
    } />
</View>
`;
